lib/rules/use-ember-data-rfc-395-imports.js

```diff
@@ -0,0 +1,176 @@
+
+const ERROR_MESSAGE =
+  'Imports from @ember-data packages should be prefered over imports from ember-data';
+
+//------------------------------------------------------------------------------
+// Rule Definition - Use "Ember Data Packages" from Ember RFC #395
+//------------------------------------------------------------------------------
+
+module.exports = {
+  meta: {
+    docs: {
+      description: 'Use "Ember Data Packages" from Ember RFC #395',
+      category: 'Ember Data',
+      recommended: false,
+      url:
+        'https://github.com/ember-cli/eslint-plugin-ember/tree/master/docs/rules/use-ember-data-rfc-395-imports.md',
+    },
+    fixable: 'code',
+    ERROR_MESSAGE,
+  },
+
+  create(context) {
+    //----------------------------------------------------------------------
+    // Variables
+    //----------------------------------------------------------------------
+
+    const message = ERROR_MESSAGE;
+
+    //----------------------------------------------------------------------
+    // Public
+    //----------------------------------------------------------------------
+
+    return {
+      /**
+       * turn `import Model from "ember-data/model"`
+       * into `import Model from "@ember-data/model"`
+       * or
+       * `import attr from "ember-data/attr"`
+       * into `import { attr } from "@ember-data/attr"`
+       */
+      ImportDeclaration(node) {
+        if (node.source.value in OLD_DATA_IMPORTS) {
+          const fix = buildFix(node, OLD_DATA_IMPORTS);
+          context.report({ node, message, fix });
+          return;
+        }
+
+        if (node.source.value.match(/^ember-data/)) {
+          context.report(node, message);
+        }
+      },
+
+      /**
+       * warn against `const { Model } = DS`
+       * and against `const { Model } = ED` (only) if ED is imported from ember-data
+       */
+      VariableDeclarator(node) {
+        if (!node.init) return;
+
+        // Filter out non-DS variable declarations
+        if (node.init.name !== 'DS' && !isInitImportedFrom(node, 'ember-data')) {
+          return;
+        }
+
+        const properties = node.id.properties;
+        // Iterate through the destructured properties and report them
+        properties.forEach(item => {
+          const key = item.key.name;
+          const match = GLOBALS[`DS.${key}`];
+          if (match) {
+            const message = buildMessage({
+              node: item,
+              customKey: key !== item.value.name ? item.value.name : null,
+              key,
+              match,
+              type: item.type,
+            });
+            context.report(item, message);
+          }
+        });
+      },
+
+      /**
+       * warn against DS usage in `name: DS.attr('string')`
+       * taken from
+       * https://github.com/ember-cli/eslint-plugin-ember/blob/v6.7.1/lib/rules/new-module-imports.js#L63
+       */
+      'MemberExpression > Identifier[name=DS]'(node) {
```

Comment:
This function is too long and complicated and should have simple helper functions extracted from it.